File: Entities/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManager.Entities
{
    public class Task
    {

        public int Id { get; set; }
        [StringLength(250)]
        [Required]
        public string Title { get; set; }
        public string Description { get; set; }
        public int Order { get; set;}
        public DateTime CreatedAt { get; set; }

        public List<Step> Steps { get; set; }
        public List<AttachedFile> AttachedFiles { get; set; }
    }
}

File: wwwroot/js/Task.js
export class Task {

    constructor(id = 0, title = "", description = "") {
        this.id = id;
        this.title = title;
        this.description = description;
    }

    set Id(value) {
        this.id = value;
    }
    get Id() {
        return this.id;
    }

    set Title(value) {
        this.title = value;
    }
    get Title() {
        return this.title;
    }

    set Description(value) {
        this.description = value;
    }
    get Description() {
        return this.description;
    }
}